client/gutenberg/extensions/slideshow/slideshow.js

```diff
@@ -0,0 +1,101 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { Component, createRef } from '@wordpress/element';
+import classnames from 'classnames';
+import { isEqual } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSwiper from './create-swiper';
+
+class Slideshow extends Component {
+	static defaultProps = {
+		effect: 'slide',
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.slideshowRef = createRef();
+		this.btnNextRef = createRef();
+		this.btnPrevRef = createRef();
+		this.paginationRef = createRef();
+	}
+
+	componentDidMount() {
+		this.swiperInstance = this.buildSwiper();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { align, effect, images } = this.props;
+		if ( ! isEqual( images, prevProps.images ) ) {
+			this.sizeSlideshow();
+		}
+		/* A change in alignment or images only needs an update */
+		if ( align !== prevProps.align || ! isEqual( images, prevProps.images ) ) {
+			this.swiperInstance.update();
+		}
+		/* A change in effect requires a full rebuild */
+		if ( effect !== prevProps.effect ) {
+			const { activeIndex } = this.swiperInstance;
+			this.swiperInstance.destroy( true, true );
+			this.swiperInstance = this.buildSwiper( activeIndex );
+		}
+	}
+
+	render() {
+		const { align, className, effect, images } = this.props;
+		const alignClassName = align ? `align${ align }` : null;
+		const classes = classnames( className, alignClassName );
+
+		return (
+			<div className={ classes } data-effect={ effect }>
+				<div className="swiper-container" ref={ this.slideshowRef }>
+					<div className="swiper-wrapper">
+						{ images.map( ( { alt, caption, id, url } ) => (
+							<figure className="swiper-slide atavist-cover-background-color" key={ id }>
+								<div className="slide-background atavist-cover-background-color" />
+								<img
```

Comment:
Mostly because I found it more semantic (images are pieces of content in our slideshows rather than styling -- such as a `div`'s background). This is a bit academic but I thought has started to become sort of a guiding principle when deciding between using `img` vs `background-image` styling. This is not an end in and of itself -- I'm hoping that more semantic code will give us less vendor lock-in and more resilience against invalidation, should we ever switch libraries.

The second reason is the somewhat optimistic yet naive assumption that whatever we can do to a `div` we can also do to an `img` with modern CSS nowadays :grimacing: -- we also still have a wrapper `div` or two that we might be able to leverage.

Granted, this is coming from someone who isn't too great at styling. I was going to seek a designer's feedback on this -- anyone who's proficient with CSS, but TBH was hoping that it wouldn't block this PR but could be done iteratively -- as long as we freeze the `save()` format prior to release, we can also revert back to the `div`, if the `img` approach really proves untenable.

Reply:
> The second reason is the somewhat optimistic yet naive assumption that whatever we can do to a div we can also do to an img with modern CSS nowadays

I share this assumption—it'll just mean a rewrite of some of the CSS. Also, the presence of a proper `<img>` element opens different avenues for the resizing logic that may not be available when using a background image. Let's stick with your approach and adjust the styling - I can look into this.